File: app/src/main/java/com/baidu/tv/player/model/MediaType.java
package com.baidu.tv.player.model;

/**
 * 媒体类型枚举
 */
public enum MediaType {
    IMAGE(1, "图片"),
    VIDEO(2, "视频"),
    ALL(3, "图片+视频");
    
    private final int code;
    private final String name;
    
    MediaType(int code, String name) {
        this.code = code;
        this.name = name;
    }
    
    public int getCode() {
        return code;
    }
    
    public int getValue() {
        return code;
    }
    
    public String getName() {
        return name;
    }
    
    public static MediaType fromCode(int code) {
        for (MediaType type : values()) {
            if (type.code == code) {
                return type;
            }
        }
        return ALL;
    }
}

File: app/src/main/java/com/baidu/tv/player/kt/model/MediaType.kt
package com.baidu.tv.player.kt.model

/**
 * 媒体类型枚举。code 与百度 API category / 播放列表过滤约定一致：
 * 1=图片，2=视频，3=图片+视频。
 */
enum class MediaType(val code: Int, val displayName: String) {
    IMAGE(1, "图片"),
    VIDEO(2, "视频"),
    ALL(3, "图片+视频"),
    AUDIO(4, "音频");

    /** value 与 code 同义，兼容旧调用。 */
    val value: Int get() = code

    companion object {
        @JvmStatic
        fun fromCode(code: Int): MediaType =
            entries.firstOrNull { it.code == code } ?: ALL
    }
}
